src/Composite/Struct/ImmutableStruct.php

```
@@ -0,0 +1,484 @@
+        }
+
+        // Validate current struct
+        foreach ($this->rules as $field => $fieldRules) {
+            if (!isset($this->data[$field])) {
+                throw new ValidationException("Field '{$field}' is required");
+            }
+            foreach ($fieldRules as $rule) {
+                $rule->validate($this->data[$field]);
+            }
+        }
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'validate'. (undefined)

(UnusedPrivateMethod)